l10n_eu_nace/migrations/12.0.2.0.0/post-migration.py

```diff
@@ -6,13 +6,44 @@
 def migrate(cr, version):
     cr.execute(
         """
-            UPDATE res_partner rp
-            SET nace_id = imd2.res_id
-            FROM ir_model_data imd1
-            LEFT JOIN ir_model_data imd2
-                ON  imd2.module = imd1.module
-                AND imd2.name = substring(imd1.name, 5)
-            WHERE imd1.name LIKE 'old_%'
-                AND rp.category_id = imd1.res_is
+        WITH imd_new AS(
+            SELECT 
+                imd_new.name AS xml_id, 
+                imd_new.res_id AS new_res_id
+            FROM ir_model_data AS imd_new
+            WHERE imd_new.name NOT LIKE 'old_%' 
+            AND imd_new.module = 'l10n_eu_nace'
+        ),
+        imd_old AS(
+            SELECT 
+                substring(imd_old.name, 5) AS xml_id, 
+                imd_old.res_id AS old_res_id
+            FROM ir_model_data AS imd_old
+            WHERE imd_old.name LIKE 'old_%' 
+            AND imd_old.module = 'l10n_eu_nace'
+        ),
+        match_old_new AS(
+            SELECT 
+                imd_new.xml_id,
+                imd_new.new_res_id,
+                imd_old.old_res_id
+            FROM imd_new
+            INNER JOIN imd_old ON imd_new.xml_id = imd_old.xml_id
+        ),
+        partner_with_old_nace AS(
+            SELECT 
+                rp.id AS part_id,
+                rp.name,
+                rpc.category_id,
+                match_old_new.new_res_id
+            FROM res_partner AS rp
+            INNER JOIN res_partner_res_partner_category_rel AS rpc ON rpc.partner_id = rp.id
+            INNER JOIN match_old_new ON match_old_new.old_res_id = rpc.category_id
+            
+        )
+        UPDATE res_partner rp
+        SET nace_id = partner_with_old_nace.new_res_id
+        FROM partner_with_old_nace
+        WHERE rp.id = partner_with_old_nace.part_id;
             """
```

Comment:
Is this code basically doing the same thing as before? If so, my vote goes for reverting this change. SQL is usually hard to interpret (mentally), but it's easier to interpret 5 lines than 35.

Reply:
Hi @Yajo 
Actually the previous query was failing because :
* it was addressing a field category_id on res_partner that doesn't exists.
* it doesn't taking into account the fact that category_id is a many2many fields

Maybe I'm totally wrong on the analysis of the query but as it was failing on my pgsql I debugged this and propose something "more clear", in my opinion,

regards

Reply:
OK, this is a bit too much verbose IMHO, but after reading all closely, it's fixing the bug, so :+1: